financialaid/views.py

```diff
@@ -1,24 +1,205 @@
+)
+from financialaid.serializers import IncomeValidationSerializer
+from roles.roles import Permissions
+from ui.views import get_bundle_url
 
 
 class IncomeValidationView(CreateAPIView):
     """
     View for income validation API. Takes income and currency, then determines whether review
     is necessary, and if not, sets the appropriate tier for personalized pricing.
     """
-    serializer_class = FinancialAidSerializer
+    serializer_class = IncomeValidationSerializer
     authentication_classes = (SessionAuthentication, )
     permission_classes = (IsAuthenticated, )
 
-    def get_queryset(self):
+    def get_queryset(self):  # pragma: no cover
         """
-        Used for returning the view, which hasn't been defined yet.
+        Allows the DRF helper pages to load - not available in production
         """
         return None
+
+
+class ReviewFinancialAidView(UserPassesTestMixin, ListView):
+    """
+    View for reviewing financial aid requests.
+    Note: In the future, it may be worth factoring out the code for sorting into its own subclass of ListView
+    """
+    paginate_by = 10
+    context_object_name = "financial_aid_objects"
+    template_name = "review_financial_aid.html"
+    # If user doesn't pass test_func, raises exception instead of redirecting to login url
+    raise_exception = True
+    # Used to modify queryset and in context
+    selected_status = None
+    program = None
+    # Used for sorting
+    sort_field = None
+    sort_direction = ""
+    sort_field_info = None
+    sort_fields = {
+        "first_name": {
+            "display": "First Name"
+        },
+        "last_name": {
+            "display": "Last Name"
+        },
+        "location": {
+            "display": "Location"
+        },
+        "adjusted_cost": {
+            "display": "Adjusted Cost"
+        },
+        "reported_income": {
+            "display": "Reported Income"
+        },
+    }
+    sort_field_mappings = {
+        "first_name": "user__profile__first_name",
+        "last_name": "user__profile__last_name",
+        "location": "user__profile__city",
+        "adjusted_cost": "adjusted_price",
+        "reported_income": "income_usd",
+    }
+    default_sort_field = "first_name"
+
+    def test_func(self):
+        """
+        Validate user permissions (Analogous to permissions_classes for DRF)
+        """
+        self.program = get_object_or_404(Program, id=self.kwargs.get("program_id", None))
+        return has_object_permission(Permissions.CAN_EDIT_FINANCIAL_AID, self.request.user, self.program)
+
+    def get_context_data(self, **kwargs):
+        """
+        Gets context for view
+        """
+        context = super().get_context_data(**kwargs)
+
+        # Constants required in view
+        context["selected_status"] = self.selected_status
+        context["current_program_id"] = self.program.id
+        # Note: This implementation of retrieving a course price is a naive lookup that assumes
```

Comment:
Sorry @alvinsiu I meant that to be per program, not all course in the the entire application. 

I thought you were asking about variability within in the economics program. It's definitely the case that DEDP/economics courses will cost different from SCM courses.